README.md

```diff
@@ -1,9 +1,53 @@
+
+> **Note**: If the mount point is not provided (it could be the case that an end-user is trying to build without the board connected), then the target binary is not flashed. This is especially useful for building the examples with a CI workflow.
 >
-> To run the MCUBoot example, please refer to this [link](https://github.com/ARMmbed/mbed-os-example-ble/tree/fota-service-example/FOTA/MCUboot), and to run the Mock example, please refer to this [link](https://github.com/ARMmbed/mbed-os-example-ble/tree/fota-service-example/FOTA/Mock). The FOTA service specification overview can be found [here](https://github.com/ARMmbed/mbed-os-experimental-ble-services/blob/fota-service-github-ci/services/FOTA/docs/README.md).
+> In the case of the MCUboot example, the _"factory firmware"_ is saved and would require manual transfer when the board is indeed connected. In either case, the demonstration step would be the only part that requires manual intervention from the user.
 >
-> Please note that the documentation for these examples will be migrated soon to this repository (and may see some changes). 
+> **Important**: For now, the tool assumes the target board, toolchain, and example unless otherwise specified by the end-user. Currently, the only board and toolchain supported are `NRF52840_DK` and `GCC_ARM` respectively.
+
+## Examples
+
+Please refer to the example-specific READMEs for build and demonstration instructions.
+1. [Mock Example](Mock)
+2. [MCUboot Example](MCUboot)
+
+## Known Issues
+There are slight kludges with how the MCUboot and Mock examples are built with `fota.sh`. The former is a direct result of dependency conflicts between [mbed-tools](https://github.com/ARMmbed/mbed-tools), [pyocd](https://github.com/pyocd/pyOCD), and [mbed-os](https://github.com/ARMmbed/mbed-os). The latter is a result of issues with flashing the target binary to the connected board. The details of both are addressed below:
+
+1. **Dependency Conflicts**: The difference in version requirements of the [PyYAML](https://pyyaml.org) dependency imposed by both mbed-os and pyocd led to the creation of a temporary virtual environment; this is used in the "_creating and flashing the factory firmware_" stage of the MCUboot example build process. This is a known issue and would require changes to the `requirements.txt` file in the mbed-os to resolve.\
+   \
+   Another minor conflict that doesn't hinder the build process is one between mbed-tools and mbed-os on the version requirement of the [Click](https://click.palletsprojects.com/en/8.0.x/) dependency; mbed-tools requires that the minimum version of Click to be greater than 7.1 while mbed-os requires it to be greater than 7.0. Now, the dependencies for mbed-os are installed after those of mbed-tools, which overwrites the newer version and generates a conflict. This would (again) require changes to the requirements file in the mbed-os repository to bump up the minimum version number of Click to 7.1.
```

Comment:
Does the user need to know this? Is there an action required from him?

Reply:
No, there's no action required on the end-user's behalf. Please refer to my [comment](https://github.com/ARMmbed/mbed-os-example-ble-fota/pull/5#discussion_r683803142) above. 😃